Improve UI debugging

thomasDelaporte opened the pull request:
## PR Checklist

- [ ] All new classes have class-level documentation comments, if there are any at all
- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Other... Please describe:

## What is new 
- Open/close (toggle) of the debugging UI through keys
- Visibility of all objects saved in the CoopObjectManager class

Resolves #218 
![New Debug UI](https://media.discordapp.net/attachments/730517053387177996/947162181902958642/unknown.png?width=1440&height=662)

thomasDelaporte committed:
feat: toggle debug UI visibility

thomasDelaporte committed:
feat: add coop object manager to debug UI

Display all coop objects in the debugging interface by type and display the properties of each object in another window

garrettluskey committed:
Update MobilePartySerializer.cs

garrettluskey commented:
Freezes on client for some reason. Please fix

garrettluskey commented on source/Coop/Mod/DebugUtil/DebugUI.cs:
```diff
@@ -49,11 +53,99 @@ public void Update(TimeSpan frameTime)
+                DisplayCoopObjectManagers();
                 End();
+
+                foreach(object detailObject in this.detailsObjects)
```
Move block to independant section

garrettluskey commented on source/Coop/Mod/DebugUtil/DebugUI.cs:
```diff
@@ -49,11 +53,99 @@ public void Update(TimeSpan frameTime)
         }
         public int Priority { get; } = UpdatePriority.MainLoop.DebugUI;
 
+        private void DisplayCoopObjectManagers()
```
Add comments

garrettluskey reviewed (changes requested):
Fix on client and update code, then will be good.

thomasDelaporte committed:
fix: clean up & documents the functions

thomasDelaporte committed:
fix: remove static & clean up other debug UI methods

thomasDelaporte committed:
doc: last clean up & check on server + client

thomasDelaporte commented:
@garrettluskey As we quickly discussed, the freeze is probably due to the gamedata and not the code quoted here.

I cleaned up the class as much as possible and added a lot of documentation for the whole class. 

The problem now is that the client has no data in CoopObjectManager so no way to check if it matches on both sides. The persistence => parties tab is also empty in the client.

garrettluskey reviewed (approved):
Nice
